extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @fileoverview Delay loads custom fonts on AMP pages.
+ * Example:
+ * <code>
+ * <amp-font
+ *   font-family="My Font"
+ *   timeout="3s"
+ *   font-weight="bold"
+ *   while-loading-class="myfont-font-loading"
+ *   on-error-class="myfont-font-missing"
+ *   on-load-class="myfont-font-loaded"
+ *   layout="nodisplay">
+ * </amp-font>
+ * </code>
+ *
+ * the amp-font element's layout type is nodisplay.
+ */
+
+
+import * as style from '../../../src/style';
+import {timer} from '../../../src/timer';
+
+/** @private @const {Array.<string>} */
+const DEFAULT_FONTS_ = ['sans-serif', 'serif'];
+
+/** @private @const {string} */
+const TEST_STRING_ = 'MAxmTYklsjo190QW';
+
+/** @private @const {number} */
+const TOLERANCE_ = 2;
+
+/** @private @const {number} */
+const DEFAULT_TIMEOUT_ = 5000;
+
+/** @private @const {string} */
+const DEFAULT_WEIGHT_ = '400';
+
+/** @private @const {string} */
+const DEFAULT_VARIANT_ = 'normal';
+
+/** @private @const {string} */
+const DEFAULT_STYLE_ = 'normal';
+
+/** @private @const {string} */
+const DEFAULT_SIZE_ = 'medium';
+
+/** @private @const {function(function())}*/
+const RAF_ =
+    requestAnimationFrame ||
+    webkitRequestAnimationFrame ||
+    mozRequestAnimationFrame ||
+    msRequestAnimationFrame ||
+    function(fn) {
+      setTimeout(fn, 10);
+    };
+
+class AmpFont extends AMP.BaseElement {
+
+
+  /** @override */
+  prerenderAllowed() {
+    return true;
+  }
+
+
+  /** @override */
+  firstAttachedCallback() {
+    this.customFontFamily_ = this.element.getAttribute('font-family');
+    this.normalizedFontFamilyName_ =
+        normalizeFontFamily_(this.customFontFamily_);
+    this.customFontWeight_ =
+        this.element.getAttribute('font-weight') || DEFAULT_WEIGHT_;
+    this.customFontStyle_ =
+        this.element.getAttribute('font-style') || DEFAULT_STYLE_;
+    this.customFontVariant_ =
+        this.element.getAttribute('font-variant') || DEFAULT_VARIANT_;
+    this.customFontSize_ =
```

Comment:
Why do `font-size` and `font-style` matter? Are there some loading technicals why these might be available but not others?